### PR TITLE
test / test_data alignements

### PR DESCRIPTION
**Pull request by jrouaix**
Hello,
I just added some changes to have better DX.
I don't really know why the test_data was misaligned with actual tests.

### CODE REVIEW

**Comment by jrouaix** [comment]
well, it did not work.
@puffyCid  are the test data distributed in the release aligned with the current code/branch ? 
would you have another way to distribute test data that be updated more easily ?

**Comment by jrouaix** [comment]
According to the `.github/workflows/pullrequest.yml` file, I download the exact same file as the GH actions runner.
So why I end up with other values on my machine ? (Ubuntu 22.04)

image

**Comment by puffyCid** [comment]
> Ok, I know, `fs::read_dir()` doesn't return the files in the same order on distinct targets ... @puffyCid, do we order files by name before the iteration ? (will try this way on the next commit)

Ah i did not realize `read_dir()` ordering is dependent on the filesystem
I left a few comments that i think could help improve the tests
let me know your thoughts @jrouaix

**Comment by jrouaix** [comment]
> I left a few comments that i think could help improve the tests

I don't see your comments ...

File: src/parser.rs [hunk 1]
```diff
@@ -444,12 +443,18 @@ mod tests {
         let strings_results = collect_strings(&test_path.display().to_string()).unwrap();
         assert_eq!(strings_results.len(), 536);
         assert_eq!(strings_results[0].signature, 1719109785);
-        assert_eq!(strings_results[0].uuid, "5283D7FC2531558F2C1ACE9AF26A0F");
+        assert_eq!(strings_results[0].uuid, "B6B65F4DC53ED38FEB0DDF61809853");
```
[the remark this answers is not in the document]
Comment (jrouaix) [reply src/parser.rs:446]: I don't know.
For a "file" parser, the ordering is kinda obvious. Then for a "folder" parser, should the ordering be depending of the system ? the file names ? the files creation date ? the files modification date ? something more complex ?
I think the library could have an opinion about it.
Or perhaps leave the decision to the user.
And it should reflect in the tests
What to you think ?

File: src/parser.rs [hunk 2]
```diff
@@ -444,12 +443,18 @@ mod tests {
...
         assert_eq!(strings_results[0].entry_descriptors.len(), 2);
-        assert_eq!(strings_results[0].footer_data.len(), 48096);
+        assert_eq!(strings_results[0].footer_data.len(), 1707);
```
[the remark this answers is not in the document]
Comment (jrouaix) [reply src/parser.rs:448]: Yes why not, I just wanted to add more assertions, but we could be smarter.
We could also use snapshot testing (heard about https://crates.io/crates/insta ?)

File: src/parser.rs [hunk 3]
```diff
@@ -100,31 +100,24 @@ pub fn build_log(
...
+
+    let mut entries = paths
+        .flat_map(|path| {
+            path.inspect_err(|err| {
+                error!("[macos-unifiedlogs] Failed to get directory entry: {err:?}",)
+            })
+            .ok()
+        })
+        .collect::<Vec<_>>();
+    entries.sort_by(|a, b| a.file_name().as_os_str().cmp(b.file_name().as_os_str()));
```
[the remark this answers is not in the document]
Comment (jrouaix) [reply src/parser.rs:116]: see first comment https://github.com/mandiant/macos-UnifiedLogs/pull/28#discussion_r1844958802

File: src/parser.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (puffyCid) [reply src/parser.rs:446]: The ordering does not matter for parsing. The filenames are unique UUIDs, the UnifiedLog entries will point to the UUID for the filename we need.

The library parses all them and then looks them up by UUID when parsing the log data.
I dont think ordering them will provide a benefit to the parser.

File: src/parser.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (puffyCid) [reply src/parser.rs:448]: ive read about insta. Based on my understanding of it, I do not really see a major benefit over regular tests that can be done natively in Rust/cargo?

File: src/parser.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jrouaix) [reply src/parser.rs:446]: Yes, you're right, let's order in the test and assert some values ?

File: src/parser.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jrouaix) [reply src/parser.rs:448]: The benefit would be to be able to assert all parsed result easily.
But there no urge here, i'll update the PR now to order in the test

**Pull request merged**